code/game/machinery/doors/firedoor.dm

```diff
@@ -60,187 +40,100 @@
-		to_chat(user, "Sorry, you must remain able bodied and close to \the [src] in order to use it.")
-		return
-
-	if(alarmed && density && !access_granted)
-		to_chat(user, "<span class='warning'>Access denied.  Please wait for authorities to arrive, or for the alert to clear.</span>")
-		return
+			to_chat(user, "<span class='warning'>\The [src] is welded solid!</span>")
+			return
 
-	else
-		user.visible_message("\blue \The [src] [density ? "open" : "close"]s for \the [user].",\
-		"\The [src] [density ? "open" : "close"]s.",\
-		"You hear a beep, and a door opening.")
-		// Accountability!
-		if(!logged_users)
-			logged_users = list()
-		logged_users += users_name
+	if(istype(C, /obj/item/weapon/crowbar) || istype(C, /obj/item/weapon/twohanded/fireaxe))
+		if(istype(C, /obj/item/weapon/twohanded/fireaxe))
+			var/obj/item/weapon/twohanded/fireaxe/F = C 
+			if(!F.wielded)
+				return
 
-	var/needs_to_close = 0
-	if(density)
-		if(alarmed)
-			needs_to_close = 1
-		spawn()
+		if(density)
 			open()
-	else
-		spawn()
+		else
 			close()
 
-	if(needs_to_close)
-		spawn(50)
-			if(alarmed)
-				nextstate = CLOSED
-
-/obj/machinery/door/firedoor/attack_ai(mob/user as mob)
-	if(operating || stat & NOPOWER)
-		return //Already doing something or depowered.
-
-	if(blocked)
-		to_chat(user, "\red \The [src] is welded solid!")
+/obj/machinery/door/firedoor/attack_hand(mob/user)		
+	add_fingerprint(user)
+	if(welded || operating || stat & NOPOWER)
 		return
-
-	var/area/A = get_area_master(src)
-	ASSERT(istype(A)) // This worries me.
-	var/alarmed = A.air_doors_activated || A.fire
-
-	var/access_granted = 0
-	if(isAI(user) || isrobot(user))
-		access_granted = 1
-
-	if(access_granted == 1)
-		user.visible_message("\blue \The [src] [density ? "open" : "close"]s for \the [user].",\
-		"\The [src] [density ? "open" : "close"]s.",\
-		"You hear a beep, and a door opening.")
-
-	var/needs_to_close = 0
 	if(density)
-		if(alarmed)
-			needs_to_close = 1
-		spawn()
-			open()
+		open()
 	else
-		spawn()
-			close()
+		close()
+		
+/obj/machinery/door/firedoor/attack_ai(mob/user)
+	attack_hand(user)
 
-	if(needs_to_close)
-		spawn(50)
-			if(alarmed)
-				nextstate = CLOSED
+/obj/machinery/door/firedoor/attack_alien(mob/user)
+	add_fingerprint(user)
+	if(welded)
+		to_chat(user, "<span class='warning'>[src] refuses to budge!</span>")
+		return
+	open()
 
 /obj/machinery/door/firedoor/do_animate(animation)
 	switch(animation)
 		if("opening")
 			flick("door_opening", src)
 		if("closing")
 			flick("door_closing", src)
-	return
 
 /obj/machinery/door/firedoor/update_icon()
 	overlays.Cut()
 	if(density)
 		icon_state = "door_closed"
-		if(blocked)
+		if(welded)
 			overlays += "welded"
 	else
 		icon_state = "door_open"
-		if(blocked)
+		if(welded)
 			overlays += "welded_open"
-	return
 
 /obj/machinery/door/firedoor/open()
-	..()
+	. = ..()
 	latetoggle()
-	return
 
 /obj/machinery/door/firedoor/close()
-	..()
-	if(locate(/mob/living) in get_turf(src))

```

Comment:
Firedoors will now close when someone's inside of them (not sure why this was in). Can always be reverted if people disagree.